experimental/cache_size.cc

```diff
@@ -0,0 +1,65 @@
+#include <stdlib.h>
+#include <time.h>
+
+#include <algorithm>
+#include <iostream>
+
+int64_t MAX_SIZE = 16 * 1024 * 1024;
+int64_t MIN_SIZE = 1024;
+int64_t ACCESSES_NUM = 64 * 1024 * 1024;
+int REPEATS = 50;
+
+clock_t whack_cache(const int64_t sz, int64_t* accesses) {
+  char* buf = new char[sz];
+
+  clock_t start = clock();
+  // TODO fencing
+  for (int64_t i = 0; i < ACCESSES_NUM; i++) {
+    ++buf[(accesses[i] * 64) % sz];
+  }
+  // TODO: elapsed time generates zero
+  clock_t elapsed = clock() - start;
+
+  double cpu_time_used = ((double)elapsed) / CLOCKS_PER_SEC;
+
+  delete[] buf;
+  return elapsed;
+}
+
+void shuffle(int64_t* a) {
```

Comment:
I believe you can replace `shuffle(a)` with [`std::shuffle`](https://en.cppreference.com/w/cpp/algorithm/random_shuffle), something like:

```c++
std::shuffle(a, a + ACCESSES_NUM, std::mt19937(std::random_device()()));
```

Feel free to add a TODO and address in a followup!

Reply:
Added as a TODO.